dali/test/python/test_pipeline.py

```diff
@@ -146,6 +148,53 @@ def define_graph(self):
     for _ in range(n_iters):
         out = pipe.run()
 
+
+def test_multiple_input_sets_error():
+    batch_size = 128
+    class Pipe(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
+            super(Pipe, self).__init__(batch_size, num_threads, device_id)
+            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
+            self.decode = ops.ImageDecoder(device = "cpu", output_type = types.RGB)
+            self.res = ops.Resize(device="cpu", resize_x=224, resize_y=224)
+
+        def define_graph(self):
+            inputs, labels = self.input(name="Reader")
+            assert_raises(TypeError, self.decode, [inputs, inputs])
+            return (inputs)
+
+    pipe = Pipe(batch_size=batch_size, num_threads=1, device_id = 0, num_gpus = 1)
+    pipe.build()
+    out = pipe.run()
+
+def test_use_twice():
+    batch_size = 128
+    class Pipe(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
+            super(Pipe, self).__init__(batch_size, num_threads, device_id)
+            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
+            self.decode = ops.ImageDecoder(device = "cpu", output_type = types.RGB)
+            self.res = ops.Resize(device="cpu", resize_x=224, resize_y=224)
+
+        def define_graph(self):
+            inputs, labels = self.input(name="Reader")
+            images = self.decode(inputs)
+            images0 = self.res(images)
```

Comment:
It's not a comment for this PR but rather for the new approach to multiple input sets:
I think these two instances should share logical ids (and therefore seeds).

Reply:
Those calls to `self.res`? We can discuss this, but I am not sure if we want to do it. Now you can use an operator with implicitly set different seeds so it will result in different outputs every call.
This can be seen as rather obscure implementation detail or a feature.